userspace/engine/filter_evttype_resolver.cpp

```diff
@@ -47,12 +44,12 @@ void filter_evttype_resolver::visitor::evttypes(const std::string& evtname, falc
 {
 	// Fill in from 2 to PPM_EVENT_MAX-1. 0 and 1 are excluded as
 	// those are PPM_GENERIC_E/PPME_GENERIC_X
-	const struct ppm_event_info* etable = g_infotables.m_event_info;
+	static sinsp s_inspector;
```

Comment:
This `static` declaration causes a segmentation fault at closing time which is why tests are failing :/ 

```
sudo ./userspace/falco/falco -c ../falco.yaml -r ../rules/falco_rules.yaml -e ./test/traces-negative/docker-compose.scap -o json_output=True -o json_include_output_property=True -o json_include_tags_property=True -o priority=debug -v
Tue Oct 11 22:54:09 2022: Falco version: 0.32.1-256+8cfaf29 (x86_64)
Tue Oct 11 22:54:09 2022: Falco initialized with configuration file: ../falco.yaml
Tue Oct 11 22:54:09 2022: Loading rules from file ../rules/falco_rules.yaml
Tue Oct 11 22:54:09 2022: Reading system call events from file: ./test/traces-negative/docker-compose.scap
{"hostname":"ubuntu2204.localdomain","output":"18:13:54.349506519: Notice Redirect stdout/stdin to network connection (user=<NA> user_loginuid=-1 examples_postgres_1 (id=d81a586c3db9) process=sh parent=initdb cmdline=sh -c \"/usr/lib/postgresql/9.5/bin/postgres\" --boot -x0 -F -c max_connections=100 -c shared_buffers=1000 -c dynamic_shared_memory_type=none < \"/dev/null\" > \"/dev/null\" 2>&1 pid=22106 terminal=0 container_id=d81a586c3db9 image=<NA> fd.name=10.0.2.15:55336->52.203.73.197:443 fd.num=1 fd.type=ipv4 fd.sip=52.203.73.197)","priority":"Notice","rule":"Redirect STDOUT/STDIN to Network Connection in Container","source":"syscall","tags":[],"time":"2016-05-24T18:13:54.349506519Z", "output_fields": {"container.id":"d81a586c3db9","container.image.repository":null,"container.name":"examples_postgres_1","evt.time":1464113634349506519,"fd.name":"10.0.2.15:55336->52.203.73.197:443","fd.num":1,"fd.sip":"52.203.73.197","fd.type":"ipv4","proc.cmdline":"sh -c \"/usr/lib/postgresql/9.5/bin/postgres\" --boot -x0 -F -c max_connections=100 -c shared_buffers=1000 -c dynamic_shared_memory_type=none < \"/dev/null\" > \"/dev/null\" 2>&1","proc.name":"sh","proc.pid":22106,"proc.pname":"initdb","proc.tty":0,"user.loginuid":-1,"user.name":"<NA>"}}
{"hostname":"ubuntu2204.localdomain","output":"18:13:54.445505588: Notice Redirect stdout/stdin to network connection (user=<NA> user_loginuid=-1 examples_postgres_1 (id=d81a586c3db9) process=sh parent=initdb cmdline=sh -c \"/usr/lib/postgresql/9.5/bin/postgres\" --boot -x0 -F -c max_connections=100 -c shared_buffers=16384 -c dynamic_shared_memory_type=none < \"/dev/null\" > \"/dev/null\" 2>&1 pid=22108 terminal=0 container_id=d81a586c3db9 image=<NA> fd.name=10.0.2.15:55336->52.203.73.197:443 fd.num=1 fd.type=ipv4 fd.sip=52.203.73.197)","priority":"Notice","rule":"Redirect STDOUT/STDIN to Network Connection in Container","source":"syscall","tags":[],"time":"2016-05-24T18:13:54.445505588Z", "output_fields": {"container.id":"d81a586c3db9","container.image.repository":null,"container.name":"examples_postgres_1","evt.time":1464113634445505588,"fd.name":"10.0.2.15:55336->52.203.73.197:443","fd.num":1,"fd.sip":"52.203.73.197","fd.type":"ipv4","proc.cmdline":"sh -c \"/usr/lib/postgresql/9.5/bin/postgres\" --boot -x0 -F -c max_connections=100 -c shared_buffers=16384 -c dynamic_shared_memory_type=none < \"/dev/null\" > \"/dev/null\" 2>&1","proc.name":"sh","proc.pid":22108,"proc.pname":"initdb","proc.tty":0,"user.loginuid":-1,"user.name":"<NA>"}}
Elapsed time: 0.959, Captured Events: 457171, 476897.88 eps
Syscall event drop monitoring:
   - event drop detected: 0 occurrences
   - num times actions taken: 0
Events detected: 2
Rule counts by severity:
   NOTICE: 2
Triggered rules by rule name:
   Redirect STDOUT/STDIN to Network Connection in Container: 2
[1]    40017 segmentation fault  sudo ./userspace/falco/falco -c ../falco.yaml -r ../rules/falco_rules.yaml -e
````

Reply:
Using a unique pointer fixes it but probably we need to find another solution

```c
std::unique_ptr<sinsp> inspector(new sinsp());
const auto etable = inspector->get_event_info_tables()->m_event_info;
```

Reply:
You're right. Let's rollback to using the global table and think about adding static methods in sinsp to retrieve them.